src/commands/amend.rs

```diff
@@ -0,0 +1,63 @@
+//! Amend the current commit.
+//!
+//! This command amends the HEAD commit with changes to files
+//! that are already tracked in the repo. Following the amend,
+//! the command performs a restack.
+
+use std::time::SystemTime;
+
+use eyre::Context;
+
+use crate::commands::gc::mark_commit_reachable;
+use crate::commands::restack;
+use crate::core::eventlog::{Event, EventLogDb};
+use crate::git::{GitRunInfo, Repo};
+use crate::opts::MoveOptions;
+use crate::tui::Effects;
+
+/// Amends the existing HEAD commit.
+pub fn amend(
+    effects: &Effects,
+    git_run_info: &GitRunInfo,
+    move_options: &MoveOptions,
+) -> eyre::Result<isize> {
+    let now = SystemTime::now();
+    let repo = Repo::from_current_dir()?;
+    let conn = repo.get_db_conn()?;
+    let mut event_log_db = EventLogDb::new(&conn)?;
+
+    let head_oid = match repo.get_head_info()?.oid {
+        Some(oid) => oid,
+        None => eyre::bail!("No HEAD to amend to"),
+    };
+    let head_commit = repo.find_commit_or_fail(head_oid)?;
+
+    let mut index = repo.get_index()?;
+    // If the repo has staged changes, only commit those. If there
+    // are no staged changes, update all files from the working tree.
+    if !repo.has_staged_changes()? {
+        index.update_all(&["*"])?;
```

Comment:
We can't use `index.update_all` because `libgit2` won't use the `fsmonitor` system in use for large repositories. You really do have to shell out to a Git command in order to ensure that you use it. If you call `git status`, it will summarize both staged and unstaged changes in the same operation, and it will also update the index on disk, so you won't need to update it yourself.